nodes/vector/turbulence.py

```diff
@@ -108,22 +110,49 @@ def process(self):
         m_hard = inputs['Hard'].sv_get()[0][0]
         m_amp = inputs['Amplitude'].sv_get()[0][0]
         m_freq = inputs['Frequency'].sv_get()[0][0]
+        rseed = inputs['Random seed'].sv_get()[0][0]
+
 
 
         _noise_type = noise_dict[self.noise_type]
         turbulence_function = turbulence_f[self.out_mode]
 
-
         if verts and verts[0]:
-            for vertices in verts:
-                out.append([turbulence_function(v, m_octaves, m_hard, _noise_type, m_amp, m_freq) for v in vertices])
-        
+            for i,vertices in enumerate(verts[0]):
+                print('verts: {0}'.format(verts))
+                print(vertices)
+                print(vertices[0])
+                coords = []
+                print('index: {0}'.format(i))
+                # origin
+                if rseed == 0:
+                    origin = 0.0,0.0,0.0
+                    origin_x = 0.0
+                    origin_y = 0.0
+                    origin_z = 0.0
+                else:
+                # randomise origin
+                    noise.seed_set( rseed )
+                    #origin = noise.random_unit_vector(3)
+                    #print('origin is: {0}'.format(origin))
+                    #origin_x = ( 0.5 - origin[0] ) * 20.0
+                    #origin_y = ( 0.5 - origin[1] ) * 20.0
+                    #origin_z = ( 0.5 - origin[1] ) * 20.0
+                    origin_x = noise.random()  * 20.0
+                    origin_y = noise.random()  * 20.0
+                    origin_z = noise.random()  * 20.0
+
+                coords.append((vertices[0] / m_octaves + origin_x, vertices[1] / m_octaves + origin_y, vertices[2] / m_octaves + origin_z))
+                print('coords: {0}'.format(coords))
+                out.append([turbulence_function(v, m_octaves, m_hard, _noise_type, m_amp, m_freq) for v in coords])
```

Comment:
Since you are building up the ```coords``` but for every vertex append a new list of the ```coords``` you already have